torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -137,9 +135,9 @@ Reducer::Reducer(
   // we do not expect sparse gradients for any parameter.
   if (expect_sparse_gradients_.empty()) {
     expect_sparse_gradients_ = std::vector<std::vector<bool>>(
```

Comment:
Can be done in followup PRs: Is it possible to simplify this to an `std::vector<bool>`?

Reply:
Done